libs/cardano-ledger-core/testlib/Test/Cardano/Ledger/Core/Binary/CDDL.hs

```diff
@@ -419,3 +463,224 @@ mkScriptInvalidHereafter =
+
+drep_deregistration_cert :: Named Group
+drep_deregistration_cert = "drep_deregistration_cert" =:~ grp [17, a drep_credential, a coin]
+
+stake_delegation_cert :: Named Group
+stake_delegation_cert = "stake_delegation_cert" =:~ grp [2, a stake_credential, a pool_keyhash]
+
+vote_delegation_cert :: Named Group
+vote_delegation_cert = "vote_delegation_cert" =:~ grp [9, a stake_credential, a drep]
+
+stake_vote_delegation_cert :: Named Group
+stake_vote_delegation_cert =
+  "stake_vote_delegation_cert" =:~ grp [10, a stake_credential, a pool_keyhash, a drep]
+
+stake_registration_delegation_cert :: Named Group
+stake_registration_delegation_cert =
+  "stake_registration_delegation_cert" =:~ grp [11, a stake_credential, a pool_keyhash, a coin]
+
+vote_registration_delegation_cert :: Named Group
+vote_registration_delegation_cert =
+  "vote_registration_delegation_cert" =:~ grp [12, a stake_credential, a drep, a coin]
+
+stake_vote_registration_delegation_cert :: Named Group
+stake_vote_registration_delegation_cert =
+  "stake_vote_registration_delegation_cert"
+    =:~ grp [13, a stake_credential, a pool_keyhash, a drep, a coin]
+
+committee_authorization_cert :: Named Group
+committee_authorization_cert =
+  comment "Authorize committee hot key for cold key" $
+    "committee_authorization_cert"
+      =:~ grp [14, a committee_cold_credential, a committee_hot_credential]
+
+committee_resignation_cert :: Named Group
+committee_resignation_cert =
+  comment "Resign from committee with cold key" $
+    "committee_resignation_cert"
+      =:~ grp [15, a committee_cold_credential, a $ anchor / VNil]
+
+drep_update_cert :: Named Group
+drep_update_cert = "drep_update_cert" =:~ grp [18, a drep_credential, a $ anchor / VNil]
+
+genesis_delegation_cert :: Named Group -- Shelley through Babbage only
+genesis_delegation_cert =
+  "genesis_delegation_cert"
+    =:~ grp [5, a genesis_hash, a genesis_delegate_hash, a vrf_keyhash]
+
+move_instantaneous_rewards_cert :: Named Group -- Shelley through Babbage only
+move_instantaneous_rewards_cert =
+  "move_instantaneous_rewards_cert" =:~ grp [6, a move_instantaneous_reward]
+
+move_instantaneous_reward :: Rule
+move_instantaneous_reward =
+  comment
+    [str|The first field determines where the funds are drawn from.
+        |  0 denotes the reserves,
+        |  1 denotes the treasury.
+        |If the second field is a map, funds are moved to stake credentials.
+        |Otherwise, the funds are given to the other accounting pot.
+        |NOTE:
+        |  This has been safely backported to Shelley from Alonzo.
+        |]
+    $ "move_instantaneous_reward"
+      =:= arr
+        [ a (int 0 / int 1)
+        , a (smp [0 <+ asKey stake_credential ==> delta_coin] / coin)
+        ]
+
+delta_coin :: Rule
+delta_coin =
+  comment
+    [str|This too has been introduced in Shelley as a backport from Alonzo.
+        |]
+    $ "delta_coin" =:= VInt
```

Comment:
This too can go into cardano-ledger-shelley, since it is only used for the MIR certificate.